general/socketstream.cpp

```diff
@@ -108,11 +108,19 @@ int socketbuf::open(const char hostname[], int port)
    hints.ai_socktype = SOCK_STREAM;
    hints.ai_flags = 0;
    hints.ai_protocol = 0;
+   // On Windows, the following need to be set to 0; also required by POSIX.
+   hints.ai_addrlen = 0;
+   hints.ai_canonname = NULL;
+   hints.ai_addr = NULL;
+   hints.ai_next = NULL;
 
    std::string portStr = std::to_string(port);
    int s = getaddrinfo(hostname, portStr.c_str(), &hints, &res);
    if (s != 0)
    {
+#ifdef MFEM_DEBUG
+      mfem::out << "Error in getaddrinfo(): code = " << s << std::endl;
```

Comment:
```suggestion
      mfem::err << "Error in getaddrinfo(): code = " << s << std::endl;
```
Perhaps?

Reply:
Other similar debug messages in this file go to `mfem::out`, so I used that as well. We can change it here and in other places in this file, if you think it's worth switching to `mfem::err`.

Reply:
`mfem::err` seems more logical, but it's not particularly important to me.

Reply:
I also think `mfem::err` is more logical for a message reporting an error

Reply:
Okay, updated in fe8562e6.